devices/ikea.js

```diff
@@ -627,18 +628,48 @@ module.exports = [
         description: 'STARKVIND air purifier',
         exposes: [
             e.fan().withModes(['off', 'low', 'medium', 'high', 'auto']),
+            exposes.enum('ikea_fan_mode', ea.ALL, [
+                'off', 'auto', '1', '1.5', '2', '2.5', '3', '3.5', '4', '4.5', '5',
+            ]).withDescription('Mode of this fan using finer fan speed control'),
+            exposes.enum('ikea_fan_speed', ea.STATE_GET, [
```

Comment:
As you suggested in https://github.com/Koenkk/zigbee-herdsman/pull/455:

> @Koenkk input here would be welcome, I could just use a fan_mode and fan_speed state but then fan_mode will deviate from what is currently available in fan_mode and from the ZCL range, as this one would be based on the ikea cluster and not hvacFanCtrl, I kind of want... both ?

Deviating for the standard fan_mode is fine as long as it is in the `withModes([..])`, I cannot try it but https://github.com/Koenkk/zigbee2mqtt/blob/b3e9afaf3ee55e2d62f6775dc0c62e8550b8fef1/lib/extension/homeassistant.ts#L381 seems to be able to handle this.

Reply:
So if I understand correctly:
- drop fz.fan_mode, tz.fan-Mode
- update fz.ikea_air_purifier and tz.ikea_air_purifier_fan_mode and rename `ikea_fan_mode` to `fan_mode` and add `fan_state` to keep it consistant with tz.fan_mode's 2 properties
- for constancy also rename `ikea_fan_speed` property to just `fan_speed` ?

Reply:
The ikea app seems to use .5 increments but maybe we should just use 1-10 as a fan_speed/fan_mode level ? Those 'x.5' are a pain to send using mosquitto_pub depending on how they are quotes.

Reply:
- With the original bridge, does the gateway use the ZCL fan cluster at all?


> The ikea app seems to use .5 increments but maybe we should just use 1-10 as a fan_speed/fan_mode level ? Those 'x.5' are a pain to send using mosquitto_pub depending on how they are quotes.

Good idea, so basically in toZigbee do value / 2, in fromZigbee * 2

Reply:
0=off,
1=auto
10=1
15=2
20=3
25=4
30=5
35=6
40=7
45=8
50=9

Erm so 1-9, I thought there was a 55 too but there isn't.

Currently it's 10 * fan speed, so 10 * 1.5 => 15, ... although I'm using a constact that just maps it so we get off and auto in there too with just using getKey.

Reply:
>     * With the original bridge, does the gateway use the ZCL fan cluster at all?
AFAIK, know I think it's just there for compat with 3rd party gateways. It support off, auto, low, med, high according to the sequance cluster which gives access to about 1/3 of the speeds.

Reply:
> I think it's just there for compat with 3rd party gateways.

Makes sense, then this is good (just ikea specific converters)